src/gcp/cloudsql/permissions.ts

```diff
@@ -0,0 +1,267 @@
+    `GRANT EXECUTE ON ALL FUNCTIONS IN SCHEMA "${schema}" TO "${firebaseWriterRole}"`,
+
+    // Set reader defaults for new tables
+    `SET ROLE = '${firebaseowner(databaseId)}';`,
+    `ALTER DEFAULT PRIVILEGES IN SCHEMA "${schema}" GRANT SELECT, INSERT, UPDATE, DELETE, TRUNCATE ON TABLES TO "${firebaseWriterRole}";`,
+    `ALTER DEFAULT PRIVILEGES IN SCHEMA "${schema}" GRANT USAGE ON SEQUENCES TO "${firebaseWriterRole}";`,
+    `ALTER DEFAULT PRIVILEGES IN SCHEMA "${schema}" GRANT EXECUTE ON FUNCTIONS TO "${firebaseWriterRole}"`,
+    `SET ROLE = cloudsqlsuperuser`,
+  ];
+}
+
+// The SQL permissions required for a role to read the FDC databases.
+// Requires the firebase_owner_* role to be the owner of the schema for default permissions.
+function readerRolePermissions(databaseId: string, superuser: string, schema: string): string[] {
+  const firebaseReaderRole = firebasereader(databaseId);
+  return [
+    `do
+      $$
+      begin
+        if not exists (select FROM pg_catalog.pg_roles
+          WHERE  rolname = '${firebaseReaderRole}') then
+          CREATE ROLE "${firebaseReaderRole}" WITH ADMIN "${superuser}";
+        end if;
+      end
+      $$
+    ;`,
+
+    `GRANT "${firebaseReaderRole}" TO "cloudsqlsuperuser"`,
+
+    `GRANT USAGE ON SCHEMA "${schema}" TO "${firebaseReaderRole}"`,
+
+    `GRANT SELECT ON ALL TABLES IN SCHEMA "${schema}" TO "${firebaseReaderRole}"`,
+
+    // Grant reader usage on sequences for nextval()
+    `GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA "${schema}" TO "${firebaseReaderRole}"`,
+
+    // Grant execution on function which could be needed by some extensions.
+    `GRANT EXECUTE ON ALL FUNCTIONS IN SCHEMA "${schema}" TO "${firebaseReaderRole}"`,
+
+    // Set reader defaults for new tables.
+    // Only the owner of the schema can set defaults.
+    `SET ROLE = '${firebaseowner(databaseId)}';`,
+    `ALTER DEFAULT PRIVILEGES IN SCHEMA "${schema}" GRANT SELECT ON TABLES TO "${firebaseReaderRole}";`,
+    `ALTER DEFAULT PRIVILEGES IN SCHEMA "${schema}" GRANT SELECT, USAGE ON SEQUENCES TO "${firebaseReaderRole}";`,
+    `ALTER DEFAULT PRIVILEGES IN SCHEMA "${schema}" GRANT EXECUTE ON FUNCTIONS TO "${firebaseReaderRole}"`,
+    `SET ROLE = cloudsqlsuperuser`,
+  ];
+}
+
+// Sets up all FDC roles (owner, writer, and reader).
+// Creates the P4SA database user and grants it writer role.
+// Returns the superuser name and password if further setup is needed by caller (e.g Granting roles to users).
+export async function setupSQLPermissions(
+  instanceId: string,
+  databaseId: string,
+  options: Options,
+  silent: boolean = false,
+) {
+  const projectId = needProjectId(options);
+
+  // 1. Create a temporary builtin user
+  const superuser = "firebasesuperuser";
+  const temporaryPassword = utils.generateId(20);
+  await cloudSqlAdminClient.createUser(
+    projectId,
+    instanceId,
+    "BUILT_IN",
+    superuser,
+    temporaryPassword,
+  );
```

Comment:
I think `executeSqlCmdsAsSuperUser` below already does this.

Reply:
True, I forgot to remove this.